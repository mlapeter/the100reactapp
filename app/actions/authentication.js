export const FETCH_TOKEN = "FETCH_TOKEN";
export const FETCH_TOKEN_RESULT = "FETCH_TOKEN_RESULT";
export const FETCH_TOKEN_ERROR = "FETCH_TOKEN_ERROR";

export const DECODE_TOKEN = "DECODE_TOKEN";
export const DECODE_TOKEN_RESULT = "DECODE_TOKEN_RESULT";
export const DECODE_TOKEN_ERROR = "DECODE_TOKEN_ERROR";
export const REMOVE_TOKEN = "REMOVE_TOKEN";
export const REMOVE_TOKEN_ERROR = "REMOVE_TOKEN_ERROR";
export const SET_FIREBASE_TOKEN = "SET_FIREBASE_TOKEN";
export const RESET_PASSWORD = "RESET_PASSWORD";
export const RESET_PASSWORD_RESULT = "RESET_PASSWORD_RESULT";
export const RESET_PASSWORD_ERROR = "RESET_PASSWORD_ERROR";

export const fetchToken = (username, password, temp_auth_token) => ({
  type: FETCH_TOKEN,
  username,
  password,
  temp_auth_token
});

export const decodeToken = token => ({
  type: DECODE_TOKEN,
  token
});

export const removeToken = () => ({
  type: REMOVE_TOKEN
});

export const setFirebaseToken = token => ({
  type: SET_FIREBASE_TOKEN,
  token
});

export const resetPassword = resetPasswordEmail => ({
  type: RESET_PASSWORD,
  resetPasswordEmail
});
