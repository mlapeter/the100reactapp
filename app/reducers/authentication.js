import {
  FETCH_TOKEN,
  FETCH_TOKEN_RESULT,
  FETCH_TOKEN_ERROR,
  DECODE_TOKEN,
  DECODE_TOKEN_RESULT,
  DECODE_TOKEN_ERROR,
  REMOVE_TOKEN,
  REMOVE_TOKEN_ERROR,
  SET_FIREBASE_TOKEN
} from "../actions/authentication";

const initialState = {
  appLoading: true,
  isLoading: false,
  isAuthed: false,
  user: {},
  token: "",
  firebaseToken: ""
};

export default (state = initialState, action) => {
  switch (action.type) {
    case FETCH_TOKEN:
      return {
        ...state,
        username: action.username,
        password: action.password,
        isLoading: true
      };
    case FETCH_TOKEN_RESULT:
      return {
        ...state,
        token: action.token,
        firebaseToken: action.firebaseToken
      };
    case FETCH_TOKEN_ERROR:
      return {
        ...state,
        error: action.error,
        errorAt: new Date(),
        isLoading: false,
        isAuthed: false,
        token: "",
        firebaseToken: ""
      };
    case DECODE_TOKEN:
      return {
        ...state,
        token: action.token,
        isLoading: true
      };
    case DECODE_TOKEN_RESULT:
      return {
        ...state,
        user: action.result,
        isLoading: false,
        isAuthed: true
      };
    case DECODE_TOKEN_ERROR:
      return {
        ...state,
        error: action.error,
        errorAt: new Date(),
        isLoading: false,
        isAuthed: false
      };
    case REMOVE_TOKEN:
      return {
        ...state,
        isAuthed: false,
        isLoading: false,
        user: {},
        token: "",
        firebaseToken: ""
      };
    case REMOVE_TOKEN_ERROR:
      return {
        ...state,
        error: action.error,
        errorAt: new Date(),
        isLoading: false,
        isAuthed: false,
        user: {},
        token: ""
      };
    case SET_FIREBASE_TOKEN:
      return {
        ...state,
        firebaseToken: action.token
      };
    default:
      return state;
  }
};
