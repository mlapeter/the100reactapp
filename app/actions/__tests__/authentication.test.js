import { fetchToken } from "../authentication";

describe("fetchToken", () => {
  it("creates a properly formatted action", () => {
    const expected = {
      type: "FETCH_TOKEN",
      username: "user1",
      password: "12345"
    };
    const actual = fetchToken("user1", "12345");
    expect(actual).toEqual(expected);
  });

  it("creates a properly formatted action", () => {
    expect(fetchToken("user1", "12345")).toMatchSnapshot();
  });
});
