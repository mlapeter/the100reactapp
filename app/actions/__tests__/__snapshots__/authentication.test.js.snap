// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`fetchToken creates a properly formatted action 1`] = `
Object {
  "password": "12345",
  "type": "FETCH_TOKEN",
  "username": "user1",
}
`;
